go/vt/vttablet/tabletmanager/vreplication/vcopier_copy_all.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vreplication
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	"vitess.io/vitess/go/vt/sqlparser"
+
+	"google.golang.org/protobuf/encoding/prototext"
+
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/binlog/binlogplayer"
+	"vitess.io/vitess/go/vt/log"
+	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+/*
+This file is similar to vcopier.go: it handles the copy phase for the ForeignKeyMode where all tables
+are streamed in a single phase.
+*/
+
+type copyAllState struct {
+	vc               *vcopier
+	plan             *ReplicatorPlan
+	currentTableName string
+	tables           map[string]bool
+}
+
+func newCopyAllState(vc *vcopier) (*copyAllState, error) {
+	state := &copyAllState{
+		vc: vc,
+	}
+	plan, err := buildReplicatorPlan(vc.vr.source, vc.vr.colInfoMap, nil, vc.vr.stats)
+	if err != nil {
+		return nil, err
+	}
+	state.plan = plan
+	state.tables = make(map[string]bool, len(plan.TargetTables))
+	for _, table := range plan.TargetTables {
+		state.tables[table.TargetName] = false
+	}
+	return state, nil
+}
+
+func (vc *vcopier) copyAll(ctx context.Context, settings binlogplayer.VRSettings) error {
+	var err error
+
+	log.Infof("Starting copyAll for %s", settings.WorkflowName)
+	defer log.Infof("Returning from copyAll for %s", settings.WorkflowName)
+	defer vc.vr.dbClient.Rollback()
+
+	state, err := newCopyAllState(vc)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, copyPhaseDuration)
```

Comment:
We might want to use a longer timeout since we have to copy everything within this timeout, right?